### DISCUSSION TITLE
Would it be possibe to add a feature to only use a part of the figure title for cross-reference in the Table of Figures ?

### DISCUSSION THREAD

**Discussion by Minh-AnhHuynh** [quarto-dev/quarto-cli#3769] [Q&A]
Currently, it is not possible to select only part of a figure legend in order to use it as cross-reference in the Table of Figures in PDF format.

So far, the only existing implementation that I know is some pretty wordy latex commands, and it defeats the entire purpose of using visual editor in RStudio, now I have to look at ugly latex code.

For reference, here is the latex code needed to use part of a caption for cross-reference in the Table of Figure in a PDF document:

```latex
\begin{figure}
\includegraphics{figures/figure-stuff.png} 
\caption[\textbf{Select your main title}]{Main title. Add more figure description here. [@your.reference]}
\label{fig:figure-stuff}
\end{figure}
```

However there is another downside to this: you can no longer use citeproc inside the latex code. So if you want to use a Table of Figures, you will likely not be able to use citeproc too. But only citeproc allows you to see your references in visual mode using RStudio, which is very convenient. 

I understand that a Table of Figure might not be needed everytime, but currently for writing a thesis, it does not suit all of my needs.

Cheers,
Minh-Anh

**Comment by mcanouil** [comment]
I believe the answer is in Pandoc: <https://github.com/jgm/pandoc/issues/2417>, especially the follow-up opened issue <https://github.com/jgm/pandoc/issues/7915>.

**Reply by Minh-AnhHuynh** [reply]
Thank you for the links, I have checked them and it seems [short captions ](https://github.com/pandoc/lua-filters/tree/master/short-captions) is what I need. However, to make the filter work, lof:true needs to be activated, however doing so would put it on the first page with no way to move it, unless I am not aware of something, so the solution doesn't work for me unfortunately.

**Reply by mcanouil** [reply]
At some point it seems you want a template thus I suggest for you to take a look at Quarto extension template to build yours and/or partials to customise parts of your target format.

**Reply by Minh-AnhHuynh** [reply]
I have come back to this reply, to my surprise, it worked. I don't know why it didn't work before. However I encountered a new issue. I am unable to use the regular @fig-caption crossreference syntax. The title just won't work. So in the end I still have to use the latex command, which is less than ideal because doing `**(Figure \ref{fig-cap})**` in source mode modifies my text to (**Figure** \ref{fig-cap}) when going to visual mode which is not what I want. If you accidentally hit the visual mode, all your Figure references format get modified which is very annoying.

In the end I replace the huge latex block with a nicer pandoc syntax, however crossreferencing it is still a pain.

**Reply by mcanouil** [reply]
Without a complete reproducible example of your (new) issue, it's still hard (actually nearly impossible) to help you.
It might be a bug, a mis-use or anything else really.

So, please provide a complete but small Quarto document that exhibit your issue if you want me or others to help. 😉
